scipy/interpolate/_bsplines.py

```diff
@@ -618,9 +620,21 @@ def make_interp_spline(x, y, k=3, t=None, bc_type=None, axis=0,
     if k == 0:
         if any(_ is not None for _ in (t, deriv_l, deriv_r)):
             raise ValueError("Too much info for k=0.")
+        x = _as_float_array(x, check_finite)
         t = np.r_[x, x[-1]]
-        c = y
-        return BSpline(t, c, k, axis=axis)
+        c = np.asarray(y)
+        c = np.ascontiguousarray(c, dtype=_get_dtype(c.dtype))
+        return BSpline.construct_fast(t, c, k, axis=axis)
+
+    # special-case k=1 (e.g., Lyche and Morken, Eq.(2.16))
+    if k == 1 and t is None:
+        if not (deriv_l is None and deriv_r is None):
+            raise ValueError("Too much info for k=1.")
+        x = _as_float_array(x, check_finite)
+        t = np.r_[x[0], x, x[-1]]
+        c = np.asarray(y)
+        c = np.ascontiguousarray(c, dtype=_get_dtype(c.dtype))
```

Comment:
Instead of two potentially memcopy-like steps, can you use a numpy function to get the dtype, and then do the memcopy-like operation?